### PR TITLE
Attempt at creating a good docker compose file for production and development

### PR DESCRIPTION
**Pull request by Koen1999**
Inspired by #988

Still not quite perfect...

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    && docker-php-ext-configure intl \
+    && docker-php-ext-install \
+        calendar \
+        exif \
+        gd \
+        intl \
+        mbstring \
+        pdo \
+        pdo_mysql \
+        soap \
```
Comment (jszanto) [inline Dockerfile:32]: No longer needed

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,59 @@
+FROM node:14-alpine as node-build
+WORKDIR /code
+
+ADD package.json package-lock.json ./
+RUN npm install
+
+ADD public/ ./public/
+RUN npm run scss
+
+FROM php:5.6-fpm as php-target
```
Comment (jszanto) [inline Dockerfile:10]: Consider using the alpine php image, it is a lot smaller

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        libicu-dev \
+        libjpeg62-turbo-dev \
+        libmagickwand-dev \
+        libpng-dev \
+        libxml2-dev \
+        unzip \
+        zlib1g-dev \
+    && docker-php-ext-configure gd --with-freetype-dir=/usr/include/ --with-jpeg-dir=/usr/include/ \
+    && docker-php-ext-configure intl \
+    && docker-php-ext-install \
```
Comment (jszanto) [inline Dockerfile:24]: Don't forget to add something like opcache for production

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,59 @@
+FROM node:14-alpine as node-build
```
Comment (jszanto) [inline Dockerfile:1]: I would recommend splitting this in two docker files. Or use stages such that you have a development stage which still includes all the tools such as npm and composer.

File: build.sh [hunk 5]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
```
Comment (jszanto) [inline build.sh:1]: This can be done in the dockerfile, doesn't need to be a seperate bash script

**Review by jszanto — CHANGES_REQUESTED** [review]
General remarks:
- Consider putting many configuration options, such as database credentials in env variables for production
- Why did the permissions on so many files change? Can you revert that?

File: Dockerfile (hunk 4 above)
Comment (Koen1999) [reply Dockerfile:1]: I think that the idea of the current setup is that is should be faster. I was thinking about making a docker-compose override file for development. In this I could just re-add some stuff that is useful.

File: Dockerfile (hunk 1 above)
Comment (Koen1999) [reply Dockerfile:32]: Which of the four?

**Pull request closed without merging**